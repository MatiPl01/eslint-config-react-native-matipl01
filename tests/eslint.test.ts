import { ESLint } from 'eslint';

describe('eslint', () => {
  it('returns correct configuration', async () => {
    const eslint = new ESLint({
      overrideConfigFile: './index.js'
    });

    expect(await eslint.calculateConfigForFile('test.tsx')).toMatchSnapshot();
  });
});
