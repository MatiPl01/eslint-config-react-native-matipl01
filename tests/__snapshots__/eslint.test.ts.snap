// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eslint returns correct configuration 1`] = `
{
  "languageOptions": {
    "ecmaVersion": 2018,
    "parser": "typescript-eslint/parser@7.7.0",
    "parserOptions": {
      "ecmaFeatures": {
        "jsx": true,
      },
      "project": true,
    },
    "sourceType": "module",
  },
  "linterOptions": {
    "reportUnusedDisableDirectives": 1,
  },
  "plugins": [
    "@",
    "@typescript-eslint:@typescript-eslint/eslint-plugin@7.7.0",
    "react",
    "import",
    "perfectionist:eslint-plugin-perfectionist",
    "markdown:eslint-plugin-markdown@4.0.1",
    "no-relative-import-paths",
    "prettier:eslint-plugin-prettier@5.1.3",
    "react-hooks",
    "react-native",
    "simple-import-sort:eslint-plugin-simple-import-sort@12.1.0",
    "unused-imports",
  ],
  "processor": undefined,
  "rules": {
    "@babel/object-curly-spacing": [
      0,
    ],
    "@babel/semi": [
      0,
    ],
    "@typescript-eslint/array-type": [
      2,
      {
        "default": "generic",
      },
    ],
    "@typescript-eslint/await-thenable": [
      2,
    ],
    "@typescript-eslint/ban-ts-comment": [
      2,
    ],
    "@typescript-eslint/ban-types": [
      2,
    ],
    "@typescript-eslint/block-spacing": [
      0,
    ],
    "@typescript-eslint/brace-style": [
      0,
    ],
    "@typescript-eslint/comma-dangle": [
      0,
    ],
    "@typescript-eslint/comma-spacing": [
      0,
    ],
    "@typescript-eslint/consistent-type-exports": [
      2,
    ],
    "@typescript-eslint/consistent-type-imports": [
      2,
    ],
    "@typescript-eslint/func-call-spacing": [
      0,
    ],
    "@typescript-eslint/indent": [
      0,
    ],
    "@typescript-eslint/key-spacing": [
      0,
    ],
    "@typescript-eslint/keyword-spacing": [
      0,
    ],
    "@typescript-eslint/lines-around-comment": [
      0,
    ],
    "@typescript-eslint/member-delimiter-style": [
      0,
    ],
    "@typescript-eslint/member-ordering": [
      2,
      {
        "default": [
          "signature",
          "method",
          "constructor",
          "field",
        ],
      },
    ],
    "@typescript-eslint/naming-convention": [
      2,
      {
        "format": [
          "PascalCase",
        ],
        "selector": "enumMember",
      },
    ],
    "@typescript-eslint/no-array-constructor": [
      2,
    ],
    "@typescript-eslint/no-base-to-string": [
      2,
    ],
    "@typescript-eslint/no-duplicate-enum-values": [
      2,
    ],
    "@typescript-eslint/no-duplicate-type-constituents": [
      2,
    ],
    "@typescript-eslint/no-explicit-any": [
      2,
    ],
    "@typescript-eslint/no-extra-non-null-assertion": [
      2,
    ],
    "@typescript-eslint/no-extra-parens": [
      0,
    ],
    "@typescript-eslint/no-extra-semi": [
      0,
    ],
    "@typescript-eslint/no-floating-promises": [
      2,
      {
        "ignoreIIFE": true,
        "ignoreVoid": true,
      },
    ],
    "@typescript-eslint/no-for-in-array": [
      2,
    ],
    "@typescript-eslint/no-implied-eval": [
      2,
    ],
    "@typescript-eslint/no-loss-of-precision": [
      2,
    ],
    "@typescript-eslint/no-misused-new": [
      2,
    ],
    "@typescript-eslint/no-misused-promises": [
      2,
    ],
    "@typescript-eslint/no-namespace": [
      2,
    ],
    "@typescript-eslint/no-non-null-asserted-optional-chain": [
      2,
    ],
    "@typescript-eslint/no-redundant-type-constituents": [
      2,
    ],
    "@typescript-eslint/no-shadow": [
      2,
    ],
    "@typescript-eslint/no-this-alias": [
      2,
    ],
    "@typescript-eslint/no-unnecessary-type-assertion": [
      2,
    ],
    "@typescript-eslint/no-unnecessary-type-constraint": [
      2,
    ],
    "@typescript-eslint/no-unsafe-argument": [
      2,
    ],
    "@typescript-eslint/no-unsafe-assignment": [
      2,
    ],
    "@typescript-eslint/no-unsafe-call": [
      2,
    ],
    "@typescript-eslint/no-unsafe-declaration-merging": [
      2,
    ],
    "@typescript-eslint/no-unsafe-enum-comparison": [
      2,
    ],
    "@typescript-eslint/no-unsafe-member-access": [
      2,
    ],
    "@typescript-eslint/no-unsafe-return": [
      2,
    ],
    "@typescript-eslint/no-unused-vars": [
      2,
    ],
    "@typescript-eslint/no-var-requires": [
      2,
    ],
    "@typescript-eslint/object-curly-spacing": [
      0,
    ],
    "@typescript-eslint/prefer-as-const": [
      2,
    ],
    "@typescript-eslint/prefer-nullish-coalescing": [
      2,
      {
        "ignoreConditionalTests": true,
        "ignoreMixedLogicalExpressions": true,
      },
    ],
    "@typescript-eslint/quotes": [
      0,
    ],
    "@typescript-eslint/require-await": [
      2,
    ],
    "@typescript-eslint/restrict-plus-operands": [
      2,
    ],
    "@typescript-eslint/restrict-template-expressions": [
      2,
    ],
    "@typescript-eslint/semi": [
      0,
    ],
    "@typescript-eslint/space-before-blocks": [
      0,
    ],
    "@typescript-eslint/space-before-function-paren": [
      0,
    ],
    "@typescript-eslint/space-infix-ops": [
      0,
    ],
    "@typescript-eslint/triple-slash-reference": [
      2,
    ],
    "@typescript-eslint/type-annotation-spacing": [
      0,
    ],
    "@typescript-eslint/unbound-method": [
      2,
    ],
    "array-bracket-newline": [
      0,
    ],
    "array-bracket-spacing": [
      0,
    ],
    "array-element-newline": [
      0,
    ],
    "arrow-parens": [
      0,
    ],
    "arrow-spacing": [
      0,
    ],
    "babel/object-curly-spacing": [
      0,
    ],
    "babel/quotes": [
      0,
    ],
    "babel/semi": [
      0,
    ],
    "block-spacing": [
      0,
    ],
    "brace-style": [
      0,
    ],
    "camelcase": [
      2,
      {
        "allow": [],
        "ignoreDestructuring": false,
        "ignoreGlobals": false,
        "ignoreImports": false,
        "properties": "always",
      },
    ],
    "comma-dangle": [
      0,
    ],
    "comma-spacing": [
      0,
    ],
    "comma-style": [
      0,
    ],
    "computed-property-spacing": [
      0,
    ],
    "consistent-this": [
      2,
      "self",
    ],
    "constructor-super": [
      0,
    ],
    "curly": [
      0,
    ],
    "dot-location": [
      0,
    ],
    "dot-notation": [
      2,
      {
        "allowKeywords": true,
        "allowPattern": "",
      },
    ],
    "eol-last": [
      2,
      "always",
    ],
    "eqeqeq": [
      2,
    ],
    "flowtype/boolean-style": [
      0,
    ],
    "flowtype/delimiter-dangle": [
      0,
    ],
    "flowtype/generic-spacing": [
      0,
    ],
    "flowtype/object-type-curly-spacing": [
      0,
    ],
    "flowtype/object-type-delimiter": [
      0,
    ],
    "flowtype/quotes": [
      0,
    ],
    "flowtype/semi": [
      0,
    ],
    "flowtype/space-after-type-colon": [
      0,
    ],
    "flowtype/space-before-generic-bracket": [
      0,
    ],
    "flowtype/space-before-type-colon": [
      0,
    ],
    "flowtype/union-intersection-spacing": [
      0,
    ],
    "for-direction": [
      2,
    ],
    "func-call-spacing": [
      0,
    ],
    "function-call-argument-newline": [
      0,
    ],
    "function-paren-newline": [
      0,
    ],
    "generator-star": [
      0,
    ],
    "generator-star-spacing": [
      0,
    ],
    "getter-return": [
      2,
      {
        "allowImplicit": false,
      },
    ],
    "implicit-arrow-linebreak": [
      0,
    ],
    "import/default": [
      2,
    ],
    "import/export": [
      2,
    ],
    "import/named": [
      0,
    ],
    "import/namespace": [
      2,
    ],
    "import/no-duplicates": [
      1,
    ],
    "import/no-unresolved": [
      2,
    ],
    "indent": [
      0,
    ],
    "indent-legacy": [
      0,
    ],
    "jsx-quotes": [
      0,
    ],
    "key-spacing": [
      0,
    ],
    "keyword-spacing": [
      0,
    ],
    "linebreak-style": [
      0,
    ],
    "lines-around-comment": [
      2,
    ],
    "max-len": [
      0,
    ],
    "max-statements-per-line": [
      0,
    ],
    "multiline-ternary": [
      0,
    ],
    "new-cap": [
      2,
      {
        "capIsNew": true,
        "capIsNewExceptionPattern": "^Gesture\\.",
        "newIsCap": true,
        "properties": true,
      },
    ],
    "new-parens": [
      0,
    ],
    "newline-per-chained-call": [
      0,
    ],
    "no-array-constructor": [
      0,
    ],
    "no-arrow-condition": [
      0,
    ],
    "no-async-promise-executor": [
      2,
    ],
    "no-await-in-loop": [
      2,
    ],
    "no-case-declarations": [
      2,
    ],
    "no-catch-shadow": [
      2,
    ],
    "no-class-assign": [
      2,
    ],
    "no-comma-dangle": [
      0,
    ],
    "no-compare-neg-zero": [
      2,
    ],
    "no-cond-assign": [
      2,
    ],
    "no-confusing-arrow": [
      0,
    ],
    "no-console": [
      1,
      {
        "allow": [
          "warn",
          "error",
        ],
      },
    ],
    "no-const-assign": [
      0,
    ],
    "no-constant-binary-expression": [
      2,
    ],
    "no-constant-condition": [
      2,
      {
        "checkLoops": true,
      },
    ],
    "no-control-regex": [
      2,
    ],
    "no-debugger": [
      2,
    ],
    "no-delete-var": [
      2,
    ],
    "no-dupe-args": [
      2,
    ],
    "no-dupe-class-members": [
      0,
    ],
    "no-dupe-else-if": [
      2,
    ],
    "no-dupe-keys": [
      0,
    ],
    "no-duplicate-case": [
      2,
    ],
    "no-else-return": [
      2,
      {
        "allowElseIf": true,
      },
    ],
    "no-empty": [
      2,
      {
        "allowEmptyCatch": false,
      },
    ],
    "no-empty-character-class": [
      2,
    ],
    "no-empty-function": [
      2,
      {
        "allow": [
          "constructors",
        ],
      },
    ],
    "no-empty-pattern": [
      2,
    ],
    "no-empty-static-block": [
      2,
    ],
    "no-eq-null": [
      2,
    ],
    "no-ex-assign": [
      2,
    ],
    "no-extend-native": [
      2,
    ],
    "no-extra-bind": [
      2,
    ],
    "no-extra-boolean-cast": [
      2,
      {
        "enforceForLogicalOperands": false,
      },
    ],
    "no-extra-parens": [
      0,
    ],
    "no-extra-semi": [
      0,
    ],
    "no-fallthrough": [
      2,
      {
        "allowEmptyCase": false,
        "commentPattern": "",
        "reportUnusedFallthroughComment": false,
      },
    ],
    "no-floating-decimal": [
      0,
    ],
    "no-func-assign": [
      2,
    ],
    "no-global-assign": [
      2,
    ],
    "no-implied-eval": [
      2,
    ],
    "no-import-assign": [
      2,
    ],
    "no-inner-declarations": [
      2,
      "both",
    ],
    "no-invalid-regexp": [
      2,
    ],
    "no-irregular-whitespace": [
      2,
    ],
    "no-iterator": [
      2,
    ],
    "no-label-var": [
      2,
    ],
    "no-labels": [
      2,
    ],
    "no-lone-blocks": [
      2,
    ],
    "no-lonely-if": [
      2,
    ],
    "no-loop-func": [
      2,
    ],
    "no-loss-of-precision": [
      0,
    ],
    "no-misleading-character-class": [
      2,
    ],
    "no-mixed-operators": [
      0,
    ],
    "no-mixed-spaces-and-tabs": [
      0,
    ],
    "no-multi-spaces": [
      0,
    ],
    "no-multiple-empty-lines": [
      0,
    ],
    "no-new": [
      2,
    ],
    "no-new-native-nonconstructor": [
      0,
    ],
    "no-new-object": [
      2,
    ],
    "no-new-symbol": [
      0,
    ],
    "no-new-wrappers": [
      2,
    ],
    "no-nonoctal-decimal-escape": [
      2,
    ],
    "no-obj-calls": [
      2,
    ],
    "no-octal": [
      2,
    ],
    "no-octal-escape": [
      2,
    ],
    "no-promise-executor-return": [
      2,
    ],
    "no-proto": [
      2,
    ],
    "no-prototype-builtins": [
      2,
    ],
    "no-redeclare": [
      2,
      {
        "builtinGlobals": true,
      },
    ],
    "no-regex-spaces": [
      2,
    ],
    "no-relative-import-paths/no-relative-import-paths": [
      1,
      {
        "allowSameFolder": true,
        "prefix": "@",
        "rootDir": "src",
      },
    ],
    "no-reserved-keys": [
      0,
    ],
    "no-return-assign": [
      2,
      "except-parens",
    ],
    "no-script-url": [
      2,
    ],
    "no-self-assign": [
      2,
    ],
    "no-self-compare": [
      2,
    ],
    "no-sequences": [
      2,
    ],
    "no-setter-return": [
      2,
    ],
    "no-shadow": [
      0,
    ],
    "no-shadow-restricted-names": [
      2,
    ],
    "no-space-before-semi": [
      0,
    ],
    "no-spaced-func": [
      0,
    ],
    "no-sparse-arrays": [
      2,
    ],
    "no-tabs": [
      0,
    ],
    "no-template-curly-in-string": [
      2,
    ],
    "no-this-before-super": [
      0,
    ],
    "no-trailing-spaces": [
      0,
    ],
    "no-undef": [
      0,
    ],
    "no-undef-init": [
      2,
    ],
    "no-underscore-dangle": [
      2,
    ],
    "no-unexpected-multiline": [
      2,
    ],
    "no-unreachable": [
      0,
    ],
    "no-unreachable-loop": [
      2,
      {
        "ignore": [
          "DoWhileStatement",
        ],
      },
    ],
    "no-unsafe-finally": [
      2,
    ],
    "no-unsafe-negation": [
      0,
    ],
    "no-unsafe-optional-chaining": [
      2,
    ],
    "no-unused-expressions": [
      2,
      {
        "allowShortCircuit": true,
        "allowTaggedTemplates": false,
        "allowTernary": true,
        "enforceForJSX": false,
      },
    ],
    "no-unused-labels": [
      2,
    ],
    "no-unused-private-class-members": [
      2,
    ],
    "no-unused-vars": [
      0,
    ],
    "no-useless-backreference": [
      2,
    ],
    "no-useless-catch": [
      2,
    ],
    "no-useless-escape": [
      2,
    ],
    "no-var": [
      2,
    ],
    "no-void": [
      2,
      {
        "allowAsStatement": true,
      },
    ],
    "no-whitespace-before-property": [
      0,
    ],
    "no-with": [
      2,
    ],
    "no-wrap-func": [
      0,
    ],
    "nonblock-statement-body-position": [
      2,
    ],
    "object-curly-newline": [
      0,
    ],
    "object-curly-spacing": [
      0,
    ],
    "object-property-newline": [
      0,
    ],
    "one-var-declaration-per-line": [
      0,
    ],
    "operator-linebreak": [
      0,
    ],
    "padded-blocks": [
      0,
    ],
    "perfectionist/sort-array-includes": [
      2,
      {
        "ignore-case": false,
        "order": "asc",
        "spread-last": true,
        "type": "natural",
      },
    ],
    "perfectionist/sort-astro-attributes": [
      2,
      {
        "ignore-case": false,
        "order": "asc",
        "type": "natural",
      },
    ],
    "perfectionist/sort-classes": [
      2,
      {
        "groups": [
          "index-signature",
          "static-property",
          "private-property",
          "property",
          "constructor",
          [
            "get-method",
            "set-method",
          ],
          "static-method",
          "private-method",
          "static-private-method",
          "method",
          "unknown",
        ],
        "ignore-case": false,
        "order": "asc",
        "type": "natural",
      },
    ],
    "perfectionist/sort-enums": [
      2,
      {
        "ignore-case": false,
        "order": "asc",
        "partition-by-comment": false,
        "type": "natural",
      },
    ],
    "perfectionist/sort-exports": [
      0,
      {
        "ignore-case": false,
        "order": "asc",
        "type": "natural",
      },
    ],
    "perfectionist/sort-imports": [
      0,
      {
        "custom-groups": {
          "type": {},
          "value": {},
        },
        "groups": [
          "type",
          [
            "builtin",
            "external",
          ],
          "internal-type",
          "internal",
          [
            "parent-type",
            "sibling-type",
            "index-type",
          ],
          [
            "parent",
            "sibling",
            "index",
          ],
          "object",
          "unknown",
        ],
        "ignore-case": false,
        "internal-pattern": [
          "~/**",
        ],
        "newlines-between": "always",
        "order": "asc",
        "type": "natural",
      },
    ],
    "perfectionist/sort-interfaces": [
      2,
      {
        "groups": [],
        "ignore-case": false,
        "optionality-order": "ignore",
        "order": "asc",
        "partition-by-new-line": false,
        "type": "natural",
      },
    ],
    "perfectionist/sort-intersection-types": [
      2,
      {
        "ignore-case": false,
        "order": "asc",
        "type": "natural",
      },
    ],
    "perfectionist/sort-jsx-props": [
      0,
      {
        "ignore-case": false,
        "order": "asc",
        "type": "natural",
      },
    ],
    "perfectionist/sort-maps": [
      2,
      {
        "ignore-case": false,
        "order": "asc",
        "type": "natural",
      },
    ],
    "perfectionist/sort-named-exports": [
      0,
      {
        "ignore-case": false,
        "order": "asc",
        "type": "natural",
      },
    ],
    "perfectionist/sort-named-imports": [
      0,
      {
        "ignore-case": false,
        "order": "asc",
        "type": "natural",
      },
    ],
    "perfectionist/sort-object-types": [
      0,
      {
        "ignore-case": false,
        "order": "asc",
        "type": "natural",
      },
    ],
    "perfectionist/sort-objects": [
      2,
      {
        "ignore-case": false,
        "order": "asc",
        "partition-by-comment": false,
        "partition-by-new-line": false,
        "styled-components": true,
        "type": "natural",
      },
    ],
    "perfectionist/sort-svelte-attributes": [
      2,
      {
        "ignore-case": false,
        "order": "asc",
        "type": "natural",
      },
    ],
    "perfectionist/sort-union-types": [
      2,
      {
        "ignore-case": false,
        "nullable-last": false,
        "order": "asc",
        "type": "natural",
      },
    ],
    "perfectionist/sort-vue-attributes": [
      2,
      {
        "groups": [],
        "ignore-case": false,
        "order": "asc",
        "type": "natural",
      },
    ],
    "prefer-const": [
      2,
    ],
    "prefer-rest-params": [
      2,
    ],
    "prefer-spread": [
      2,
    ],
    "prettier/prettier": [
      2,
    ],
    "quote-props": [
      0,
    ],
    "quotes": [
      0,
    ],
    "radix": [
      2,
      "as-needed",
    ],
    "react-hooks/exhaustive-deps": [
      2,
    ],
    "react-hooks/rules-of-hooks": [
      2,
    ],
    "react-native/sort-styles": [
      2,
    ],
    "react/jsx-child-element-spacing": [
      0,
    ],
    "react/jsx-closing-bracket-location": [
      0,
    ],
    "react/jsx-closing-tag-location": [
      0,
    ],
    "react/jsx-curly-brace-presence": [
      2,
      "never",
    ],
    "react/jsx-curly-newline": [
      0,
    ],
    "react/jsx-curly-spacing": [
      0,
    ],
    "react/jsx-equals-spacing": [
      0,
    ],
    "react/jsx-first-prop-new-line": [
      0,
    ],
    "react/jsx-indent": [
      0,
    ],
    "react/jsx-indent-props": [
      0,
    ],
    "react/jsx-key": [
      2,
    ],
    "react/jsx-max-props-per-line": [
      0,
    ],
    "react/jsx-newline": [
      0,
    ],
    "react/jsx-no-comment-textnodes": [
      2,
    ],
    "react/jsx-no-duplicate-props": [
      2,
    ],
    "react/jsx-no-target-blank": [
      2,
    ],
    "react/jsx-one-expression-per-line": [
      0,
    ],
    "react/jsx-props-no-multi-spaces": [
      0,
    ],
    "react/jsx-sort-props": [
      2,
      {
        "callbacksLast": true,
        "ignoreCase": true,
        "locale": "auto",
        "multiline": "last",
        "noSortAlphabetically": false,
        "shorthandFirst": false,
        "shorthandLast": true,
      },
    ],
    "react/jsx-space-before-closing": [
      0,
    ],
    "react/jsx-tag-spacing": [
      0,
    ],
    "react/jsx-wrap-multilines": [
      0,
    ],
    "react/no-children-prop": [
      2,
    ],
    "react/no-danger-with-children": [
      2,
    ],
    "react/no-deprecated": [
      2,
    ],
    "react/no-find-dom-node": [
      2,
    ],
    "react/no-is-mounted": [
      2,
    ],
    "react/no-render-return-value": [
      2,
    ],
    "react/no-unescaped-entities": [
      2,
    ],
    "react/no-unknown-property": [
      2,
    ],
    "react/no-unsafe": [
      0,
    ],
    "react/prop-types": [
      0,
    ],
    "react/react-in-jsx-scope": [
      0,
    ],
    "require-atomic-updates": [
      2,
    ],
    "require-await": [
      2,
    ],
    "require-yield": [
      2,
    ],
    "rest-spread-spacing": [
      0,
    ],
    "semi": [
      0,
    ],
    "semi-spacing": [
      0,
    ],
    "semi-style": [
      0,
    ],
    "simple-import-sort/exports": [
      2,
    ],
    "simple-import-sort/imports": [
      2,
    ],
    "sort-vars": [
      2,
      {
        "ignoreCase": false,
      },
    ],
    "space-after-function-name": [
      0,
    ],
    "space-after-keywords": [
      0,
    ],
    "space-before-blocks": [
      0,
    ],
    "space-before-function-paren": [
      0,
    ],
    "space-before-function-parentheses": [
      0,
    ],
    "space-before-keywords": [
      0,
    ],
    "space-in-brackets": [
      0,
    ],
    "space-in-parens": [
      0,
    ],
    "space-infix-ops": [
      0,
    ],
    "space-return-throw-case": [
      0,
    ],
    "space-unary-ops": [
      0,
    ],
    "space-unary-word-ops": [
      0,
    ],
    "standard/array-bracket-even-spacing": [
      0,
    ],
    "standard/computed-property-even-spacing": [
      0,
    ],
    "standard/object-curly-even-spacing": [
      0,
    ],
    "strict": [
      2,
      "never",
    ],
    "switch-colon-spacing": [
      0,
    ],
    "template-curly-spacing": [
      0,
    ],
    "template-tag-spacing": [
      0,
    ],
    "unicorn/empty-brace-spaces": [
      0,
    ],
    "unicorn/no-nested-ternary": [
      0,
    ],
    "unicorn/number-literal-case": [
      0,
    ],
    "unicorn/template-indent": [
      0,
    ],
    "unused-imports/no-unused-imports": [
      2,
    ],
    "use-isnan": [
      2,
    ],
    "valid-typeof": [
      2,
      {
        "requireStringLiterals": false,
      },
    ],
    "vue/array-bracket-newline": [
      0,
    ],
    "vue/array-bracket-spacing": [
      0,
    ],
    "vue/array-element-newline": [
      0,
    ],
    "vue/arrow-spacing": [
      0,
    ],
    "vue/block-spacing": [
      0,
    ],
    "vue/block-tag-newline": [
      0,
    ],
    "vue/brace-style": [
      0,
    ],
    "vue/comma-dangle": [
      0,
    ],
    "vue/comma-spacing": [
      0,
    ],
    "vue/comma-style": [
      0,
    ],
    "vue/dot-location": [
      0,
    ],
    "vue/func-call-spacing": [
      0,
    ],
    "vue/html-closing-bracket-newline": [
      0,
    ],
    "vue/html-closing-bracket-spacing": [
      0,
    ],
    "vue/html-end-tags": [
      0,
    ],
    "vue/html-indent": [
      0,
    ],
    "vue/html-quotes": [
      0,
    ],
    "vue/html-self-closing": [
      0,
    ],
    "vue/key-spacing": [
      0,
    ],
    "vue/keyword-spacing": [
      0,
    ],
    "vue/max-attributes-per-line": [
      0,
    ],
    "vue/max-len": [
      0,
    ],
    "vue/multiline-html-element-content-newline": [
      0,
    ],
    "vue/multiline-ternary": [
      0,
    ],
    "vue/mustache-interpolation-spacing": [
      0,
    ],
    "vue/no-extra-parens": [
      0,
    ],
    "vue/no-multi-spaces": [
      0,
    ],
    "vue/no-spaces-around-equal-signs-in-attribute": [
      0,
    ],
    "vue/object-curly-newline": [
      0,
    ],
    "vue/object-curly-spacing": [
      0,
    ],
    "vue/object-property-newline": [
      0,
    ],
    "vue/operator-linebreak": [
      0,
    ],
    "vue/quote-props": [
      0,
    ],
    "vue/script-indent": [
      0,
    ],
    "vue/singleline-html-element-content-newline": [
      0,
    ],
    "vue/space-in-parens": [
      0,
    ],
    "vue/space-infix-ops": [
      0,
    ],
    "vue/space-unary-ops": [
      0,
    ],
    "vue/template-curly-spacing": [
      0,
    ],
    "wrap-iife": [
      0,
    ],
    "wrap-regex": [
      0,
    ],
    "yield-star-spacing": [
      0,
    ],
  },
  "settings": {
    "import/extensions": [
      ".ts",
      ".cts",
      ".mts",
      ".tsx",
      ".js",
      ".jsx",
    ],
    "import/external-module-folders": [
      "node_modules",
      "node_modules/@types",
    ],
    "import/parsers": {
      "@typescript-eslint/parser": [
        ".ts",
        ".tsx",
      ],
    },
    "import/resolver": {
      "node": {
        "extensions": [
          ".js",
          ".web.js",
          ".ios.js",
          ".android.js",
        ],
      },
      "typescript": {
        "alwaysTryTypes": true,
        "project": "./tsconfig.json",
      },
    },
    "react": {
      "version": "detect",
    },
  },
}
`;
